base/strings/string.jl

```diff
@@ -85,6 +80,24 @@ function String(v::Vector{UInt8})
     return str
 end
 
+"""
+    takestring!(x) -> String
+
+Create a string from the content of `x`, emptying `x`.
+
+# Examples
+```jldoctest
+julia> v = Memory{UInt8}([0x61, 0x62, 0x63]);
+
+julia> s = takestring!(v)
+"abc"
+
+julia> isempty(v)
+true
+```
+"""
+takestring!(v::Vector{UInt8}) = String(v)
```

Comment:
Can we gain anything by having an `unsafe_takestring!(v::Vector{UInt8})` that leaves `v` unchanged, and relies on the caller not to modify it?

Reply:
Yes - I've made one locally and is about to change up all the bad uses of takestring!(::Memory) In this PR (which will now fail as I removed that method), but it'll have to wait until Sunday when I get time again :)